src/index.ts

```diff
@@ -1,923 +1,1064 @@
+          log.debug('No call_spec.args, converting value to null for zomeCall.');
+          call_spec.args = null
+        }
+
+        // NOTE: ZomeCall Structure = { 
+        // cell_id,
+        // zome_name,
+        // fn_name,
+        // payload
+        // cap,
+        // provenance
+        // }
+
+        const hosted_app_cell_id = call_spec["cell_id"];
+
+        zomeCall_response = await this.callConductor("hosted", {
+          // TODO: Determine why we can't just pass directly in the cell_id received back from appInfo call...
+          "cell_id": [Buffer.from(hosted_app_cell_id[0]), Buffer.from(hosted_app_cell_id[1])],
+          "zome_name": call_spec["zome"],
+          "fn_name": call_spec["function"],
+          "payload": call_spec["args"],
+          "cap": null, // Note: when null, this call will pass when the agent has an 'Unrestricted' status (this includes all calls to an agent's own chain)
+          "provenance": Codec.AgentId.decodeToHoloHash(agent_id),
+        });
+      } catch (err) {
+        log.error("Failed during Conductor call: %s", String(err));
+        zomeCall_response = {};
+
+        if (err.message.includes("Failed to get signatures from Client")) {
+          let new_message = anonymous === true
+            ? "Agent is not signed-in"
+            : "We were unable to contact Chaperone for the Agent signing service.  Please check ...";
+
+          log.warn("Setting error response to wormhole error message: %s", new_message);
+          holo_error = (new HoloError(new_message)).toJSON();
+        }
+        else if (err instanceof HoloError) {
+          log.warn("Setting error response to raised HoloError: %s", String(err));
+          holo_error = err.toJSON();
+        }
+        else {
+          log.fatal("Conductor call threw unknown error: %s", String(err));
+          console.error(err);
+          holo_error = {
+            "source": 'HoloError',
+            "message": err.message,
+          };
+        }
+      }
+
+      // - Servicelogger response
+      let host_response;
+
+      const host_metrics = {
+        "cpu": 7
+      };
+
+      const weblog_compat = {
+        source_ip: "100:0:0:0",
+        status_code: 200
+      }
+
+      log.debug("Form service response for signed request (%s): %s", service_signature, JSON.stringify(request, null, 4));
+      host_response = this.logServiceResponse(zomeCall_response, host_metrics, weblog_compat);
+      log.info("Service response by Host: %s", JSON.stringify(host_response, null, 4));
+
+      // Use response_id to act as waiting ID
+      const response_id = host_response.response_hash;
+      log.info("Adding service call ID (%s) to waiting list for client confirmations for agent (%s)", response_id, agent_id);
+      this.addPendingConfirmation(response_id, request, host_response, agent_id);
+
+      // - return host response
+      let response_message;
+      if (holo_error) {
```

Comment:
Can we check for an error before addPendingConfirmation? 
So that we don't trigger anything unnecessarily